src/CTA.Rules.Actions/ClassActions.cs

```diff
@@ -425,6 +428,44 @@ ClassDeclarationSyntax AddParametersToMethod(SyntaxGenerator syntaxGenerator, Cl
             return AddParametersToMethod;
         }
 
+        public Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> GetReplacePublicMethodsBodyAction(string expression)
+        {
+            ClassDeclarationSyntax ReplaceMethodModifiers(SyntaxGenerator syntaxGenerator, ClassDeclarationSyntax node)
+            {
+                var allMembers = node.Members.ToList();
+                var allMethods = allMembers.OfType<MethodDeclarationSyntax>().Where(m => m.Modifiers.Any(mod => mod.Text == "public"));
+
+                while(allMethods != null && allMethods.Count() > 0)
+                {
+                    var method = allMethods.FirstOrDefault();
+                    //ActionResult is a catch all return type
+                    MethodDeclarationActions methodActions = new MethodDeclarationActions();
+                    var addCommentActionFunc = methodActions.GetAddCommentAction("Modified to call the extracted logic.");
+                    var newMethod = addCommentActionFunc(syntaxGenerator, method);
+
+                    bool asyncCheck = method.Modifiers.Any(mod => mod.Text == "async");
+                    string returnType = asyncCheck ? "Task<ActionResult>" : "ActionResult";
+                    var newExpression = expression;
+                    if (expression.Contains("MonolithService.CreateRequest"))
+                    {
+                        newExpression = expression.Insert((asyncCheck ? expression.IndexOf("MonolithService") : expression.IndexOf("CreateRequest()")+ "CreateRequest()".Length), (asyncCheck ? "await " : ".Result"));
+                    }
+
+                    newMethod = newMethod.WithBody(null).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    newMethod = newMethod.WithBody(SyntaxFactory.Block(SyntaxFactory.ParseStatement(newExpression))).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    newMethod = newMethod.WithReturnType(SyntaxFactory.ParseTypeName(returnType)).WithLeadingTrivia(newMethod.GetLeadingTrivia());
+                    newMethod = newMethod.NormalizeWhitespace();
+
+                    node = node.ReplaceNode(method, newMethod.WithLeadingTrivia(newMethod.GetLeadingTrivia()));
+
+                    allMembers = node.Members.ToList();
+                    allMethods = allMembers.OfType<MethodDeclarationSyntax>().Where(m => m.Modifiers.Any(mod => mod.Text == "public") && !m.GetLeadingTrivia().ToFullString().Contains("Modified to call the extracted logic."));
+                }
```

Comment:
Since we're reusing a lot of these values, it might be good to move all the strings to a Constants class

Reply:
Agreed and fixed.